### DISCUSSION TITLE
[Core Protocol] Guaranteed gas market: elastic gas limit available to purchase

### DISCUSSION THREAD

**Discussion by DefiCake** [ethereum-optimism/developers#274] [Core Protocol]
### Did you check the documentation?

- [X] Yes

### Did you read the specs?

- [X] Yes
- [ ] No

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I have been reading about the [guaranteed gas market](https://github.com/ethereum-optimism/specs/blob/main/specs/protocol/guaranteed-gas-market.md), which follows an EIP1559 style auction. 

How does a dynamic gas limit `MAX_RESOURCE_LIMIT * ELASTICITY_MULTIPLIER` help at all with congestion / spam?

I understand (some of) the game theoretical implications of having limited block sizes on a L1 blockchain - latency, centralization, etc - and how a variable block size limits the feasibility of certain grieving vectors. But I cannot see how these apply to the gas market of deposit transactions for the L2. Wouldn't it be best to have a high gas limit for deposit transactions from the start, instead of incrementing it based on congestion? It would simplify the fee market, and make it harder for an attacker to price out other deposit transactions right from the start.

I cannot imagine an scenario where a spammer has it harder to grieve the L2 chain through deposit transactions with an elasticity factor vs a high base limit

**Comment by 1195597989** [comment]
L2 solutions, such as Optimistic Rollups, bundle multiple transactions into a single L1 transaction. The gas market for these deposit transactions is less about individual transaction fees and more about the cost of submitting these batches to L1. Raising the gas limit for deposit transactions could indeed streamline the fee market and potentially make it more challenging for attackers to disrupt the system by outpricing legitimate transactions. However, it’s crucial to remember that L2 chains still depend on L1 for data availability and security, so they’re not completely shielded from L1 congestion issues.
